sn_protocol/src/error.rs

```diff
@@ -110,24 +110,13 @@ pub enum Error {
     #[error("Spend parents are invalid: {0}")]
     InvalidSpendParents(String),
 
-    /// One or more parent spends of a requested spend had a different dst tx hash than the signed spend src tx hash.
-    #[error(
-        "The signed spend src tx ({signed_src_tx_hash:?}) did not match a valid parent's dst tx hash: {parent_dst_tx_hash:?}. The trail is invalid."
-    )]
-    TxTrailMismatch {
-        /// The signed spend src tx hash.
-        signed_src_tx_hash: Hash,
-        /// The dst hash of a parent signed spend.
-        parent_dst_tx_hash: Hash,
-    },
+    /// The DBC we're trying to Spend came with an invalid parent Tx
+    #[error("Invalid Parent Tx: {0}")]
+    InvalidParentTx(String),
+    /// One or more parent spends of a requested spend has an invalid hash
+    #[error("Invalid parent spend hash: {0}")]
+    BadParentSpendHash(String),
```

Comment:
Same comment as previous ones to about error types, it sounds like here the context is the parent spend hash? if there are different cases of how a parent spend hash is invalid then we may want different error types for them?

Reply:
Yeah maybe not indeed, duely noted, will fix all that in a follow up.